### PR TITLE
feat(all-pages-panel): added all pages panel

### PR DESCRIPTION
**Pull request by catakot**

### CODE REVIEW

File: src/components/AllPagesPanel/i18n/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import {registerKeyset} from '../../utils/registerKeyset';
+
+import en from './en.json';
+import ru from './ru.json';
+
+const COMPONENT = 'MobileHeader';
```
Comment (DarkGenius) [inline src/components/AllPagesPanel/i18n/index.ts:6]: Let fix component name

**Review by DarkGenius — CHANGES_REQUESTED** [review]
The create button should be at the end of the list

image

File: src/components/AllPagesPanel/AllPagesPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    onEditModeChanged?: (isEditMode: boolean) => void;
+}
+
+export const AllPagesPanel: React.FC<AllPagesPanelProps> = (props) => {
+    const {startEditIcon, onEditModeChanged, className} = props;
+    const {menuItems, onMenuItemsChanged} = useAsideHeaderInnerContext();
+    const menuItemsRef = useRef(menuItems);
+    menuItemsRef.current = menuItems;
+
+    const [isEditMode, setIsActive] = useState(false);
```
[the remark this answers is not in the document]
Comment (catakot) [reply src/components/AllPagesPanel/AllPagesPanel.tsx]: Done

**Review by Lunory — CHANGES_REQUESTED** [review]

**Comment by Lunory** [comment]
1. All pages panel doesn’t close by clicking “All pages” menu item
telegram-cloud-photo-size-2-5235718919879250331-y

2. When you are opening panel from popup, popup doesn’t close
telegram-cloud-photo-size-2-5235718919879250230-y

**Review by DarkGenius — COMMENTED** [review]
lunory@ up: _FIXED_

The element "All pages" should be highlighted while the panel is open

image

File: src/components/AsideHeader/AsideHeader.tsx [hunk 3]
```diff
@@ -1,45 +1,25 @@
...
-export interface AsideHeaderProps
-    extends AsideHeaderGeneralProps,
-        Partial<AsideHeaderDefaultProps> {}
-
-export const AsideHeader = (props: AsideHeaderInnerProps) => {
-    const {className, compact, onClosePanel} = props;
-
+export const AsideHeader = (props: AsideHeaderProps) => {
+    const {className, compact} = props;
     const size = compact ? ASIDE_HEADER_COMPACT_WIDTH : ASIDE_HEADER_EXPANDED_WIDTH;
```
[the remark this answers is not in the document]
Comment (catakot) [reply src/components/AsideHeader/AsideHeader.tsx:17]: fixed

File: src/components/AsideHeader/useAsideHeaderInnerContextValue.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        () =>
+            allPagesIsAvailable
+                ? [
+                      ...(menuItems || EMPTY_MENU_ITEMS),
+                      {
+                          ...ALL_PAGES_MENU_ITEM,
+                          onItemClick: () => {
+                              setInnerVisiblePanel(InnerPanels.AllPages);
+                          },
+                      },
```
[the remark this answers is not in the document]
Comment (catakot) [reply src/components/AsideHeader/useAsideHeaderInnerContextValue.tsx:53]: fixed

File: src/components/AsideHeader/useAsideHeaderInnerContextValue.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        if (!allPagesIsAvailable) {
+            return panelItems;
+        }
+        return [
+            ...(panelItems || []),
+            {
+                id: InnerPanels.AllPages,
+                content: <AllPagesPanel />,
+                visible: innerVisiblePanel === InnerPanels.AllPages,
+            },
```
[the remark this answers is not in the document]
Comment (catakot) [reply src/components/AsideHeader/useAsideHeaderInnerContextValue.tsx:69]: fixed

File: src/components/AllPagesPanel/AllPagesPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    }, []);
+
+    const groupedItems = useGroupedMenuItems(menuItems);
+
+    useEffect(() => {
+        onEditModeChanged?.(isEditMode);
+    }, [isEditMode, onEditModeChanged]);
+
+    const onItemClick = useCallback((item: ListItemData<MenuItem>) => {
+        //@ts-ignore TODO
```
[the remark this answers is not in the document]
Comment (catakot) [reply src/components/AllPagesPanel/AllPagesPanel.tsx]: done

File: src/components/CompositeBar/Item/Item.tsx [hunk 7]
```diff
@@ -221,7 +221,8 @@ export const Item: React.FC<ItemInnerProps> = (props) => {
     }
 
     const highlightedNode = makeIconNode(iconNode);
-
+    console.log(`open ${open}`);
```
Comment (Lunory) [inline src/components/CompositeBar/Item/Item.tsx:224]: 2 logs in this file

**Review by DarkGenius — APPROVED** [review]

**Review by Lunory — APPROVED** [review]

**Pull request merged**